apps/src/aichat/chatApi.ts

```diff
@@ -0,0 +1,71 @@
+import {ChatCompletionMessage} from './types';
+import HttpClient from '@cdo/apps/util/HttpClient';
+import {Role, Status} from './constants';
+
+export async function postOpenaiChatCompletion(
+  chatMessages: ChatCompletionMessage[]
+): Promise<ChatCompletionResponse> {
+  const payload = {messages: formatForChatCompletion(chatMessages)};
+  // Send request to chat completion backend controller.
+  const url = '/openai/chat_completion';
+  const response = await HttpClient.post(url, JSON.stringify(payload), true, {
+    'Content-Type': 'application/json; charset=UTF-8',
+  });
+  if (response.status === 200) {
+    return await response.json();
+  } else {
+    return null;
+  }
+}
+
+const formatForChatCompletion = (chatMessages: ChatCompletionMessage[]) => {
+  return chatMessages.map(message => {
+    return {role: message.role, content: message.chatMessageText};
+  });
+};
+
+export async function getChatCompletionMessage(
+  systemPrompt: string,
+  lastMessageId: number,
+  newMessage: string,
+  chatMessages: ChatCompletionMessage[]
+): Promise<ChatCompletionMessage[]> {
+  const newChatMessage: ChatCompletionMessage = {
+    id: lastMessageId + 1,
```

Comment:
Does `postOpenaiChatCompletion` need to know about other fields like id, name, status, etc, other than `role` and `chatMessageText`? I wonder if it would be better to instead convert the newMessage and systemPrompt into the format we need for the API directly, rather than first converting them into a `ChatCompletionMessage` and then into the API format. That way we're not doing any extraneous ID management and wrapping inside this function.

Reply:
Yeah - that makes sense. Will update.